src/incomingDelivery/controller.js

```diff
@@ -10,6 +13,17 @@ router.get("/", getAll);
 router.put("/", createOne);
 router.get("/queue", getQueue);
 router.post("/receive", setReceived);
+// Make sure the editMadeBy is valid
+router.post("/undoReceive", async (req, res, next) => {
+  const { editMadeBy } = req.body;
+  await checkId(res, next, User, editMadeBy);
+});
+// Make sure the deliveryId is valid
+router.post("/undoReceive", async (req, res, next) => {
```

Comment:
This doesn't need to be async as `chekcId` d/n need to be awaited when called liked this.
The router will not move to `next` until it is called inside the `checkId` routine.

Reply:
@jarrilla Good call. Do you know if this needs to be async in the TS world, just due to the signature that is expected? I cant recall off the top of my head.

Reply:
Same thing in Typescript. An async function does not need to be awaited if we don't care to run it synchronously. So if we don't have an await in the block, it doesn't need to be declared async.